docs/iam-usage.rst

```diff
@@ -0,0 +1,177 @@
+>>> policy_change.remove(iam.roles.EDITOR, [iam.domain('example.com'), iam.group('devs@example.com')])
+>>> policy, _, _ = policy_change.apply(resource)
+>>> print(iam.user('alice@example.com') in policy[iam.roles.OWNER.name])
+True
+>>> print(iam.domain('example.com') in policy[iam.roles.EDITOR.name])
+False
+
+Second the user can specify a "membership function" which will take a member string as an argument, and return ``True`` if the member should belong to the specified role, and ``False`` otherwise.
+
+>>> def membership_fn(member):
+>>>     return not iam.is_group(member) or member == iam.user('bob@example.com')
+>>> policy_change.fn(iam.roles.READER, membership_fn)
+>>> policy, _, _ policy_change.apply(resource)
+>>> print([member for member in policy[iam.roles.READER.name] if iam.is_group(member)])
+['user:bob@example.com']
+
+
+Methods
+----------------------------------
+
+Resources that implement the IAM interface provide the following methods:
+
+Low Level Methods
+~~~~~~~~~~~~~~~~~
+Resources that implement IAM provide low level methods for interacting with IAM. 
+
+``get_policy`` returns a tuple of ``(policy, version, etag)`` on the corresponding resource. 
+
+>>> policy, version, etag = resource.get_policy()
+>>> print(policy)
+{
+   'roles/owner': set(['user:alice@example.com']),
+   'roles/editor: set(['group:admins@example.com']),
+   'roles/reader': set(['domain:example.com', 'user:bob@example.com'])
+}
+>>> print(version)
+5
+>>> print(etag)
+ffdFADFdsgfsjrsHTY
+
+``set_policy`` takes a policy dictionary, as well as optional ``version`` and ``etag`` paramters. If updates are made to your policy during this change, they will be overwritten with exactly what is in your policy, or, if an etag is specified they will fail with a ``iam.ConcurrentModificationError``. ``iam.PolicyChange`` performs this "read-modify-write" cycle automatically for the user.
+
+>>> policy['roles/owner'].add('user:charles@example.com')
+>>> policy, version, etag = resource.set_policy(policy, etag=etag)
+>>> print(version)
+6
+
+``missing_permissions`` takes an iterable of "permission strings" and returns those the user does not have on the resource
+
+>>> resource.missing_permissions('resourcemanager.projects.get', 'resourcemanager.projects.delete')
+set(['resourcemanager.projects.get'])
+
+Returns permissions (if any), in the list that the user does not possess.
+
+``query_grantable_roles()`` returns a list of ``iam.Role`` objects that represent roles (and their associated metadata)
+which can be granted on the specified resource
+
+>>> resource.query_grantable_roles()
+[<Role>, <Role>, <Role>]
+
+Convenience Methods
+~~~~~~~~~~~~~~~~~~~
+
+The following methods are wrappers around the creation and application of an ``iam.PolicyChange`` object. 
+
+``add_role`` takes a single member, and a single ``iam.Role``, or role string, and adds the member to the role. ``add_role``
+
+>>> resource.add_role(iam.user('alice@example.com'), iam.roles.OWNER.name)
```

Comment:
So in essence this

1. Creates a `PolicyChange()`
1. Adds the user / role combo to the new policy change
1. Applies the new change to the `resource`

And the idea is if the resource's class inherits the mixin all will be well?